Doc/library/sys.rst

```diff
@@ -24,6 +24,54 @@ always available. Unless explicitly noted otherwise, all variables are read-only
 
    .. availability:: Unix.
 
+   .. versionchanged:: 3.14
+      A deprecation warning will be emitted if the :data:`sys.abiflags` member
+      is accessed on Windows before Python 3.16.
+      For example:
+
+      .. code-block:: python
+
+         >>> import sys
+         >>> getattr(sys, 'abiflags', None)  # on Windows
+         <python-input-1>:1: DeprecationWarning: sys.abiflags will be set to a meaningful value on all platforms ...
+         >>> hasattr(sys, 'abiflags')  # on Windows
+         <python-input-2>:1: DeprecationWarning: sys.abiflags will be set to a meaningful value on all platforms ...
+         False
+
+      To suppress this warning, use the :mod:`warnings` module:
+
+      .. code-block:: python
+
+         import warnings
+
+         with warnings.catch_warnings():
+             # ignore DeprecationWarning on sys.abiflags change on Windows
+             warnings.simplefilter('ignore', DeprecationWarning)
+             abiflags = getattr(sys, 'abiflags', '')
+
+      Due to historical reasons, :data:`sys.abiflags` is not covered by
```

Comment:
I think this all this disgression can be moved to a footnote. It's not useful for the deprecation warning itself and it makes the docs harder to read (we don't really know whether we're still inside or outside the versionchanged paragraph)

Reply:
> we don't really know whether we're still inside or outside the versionchanged paragraph

All contents are under the version-changed clause. I moved these to *What's New* anyway.

<img width="670" alt="image" src="https://github.com/user-attachments/assets/66568476-fd41-4613-9197-04f0f4269a0c" />